src/main/java/org/fisco/bcos/sdk/service/GroupManagerServiceImpl.java

```diff
@@ -0,0 +1,380 @@
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+public class GroupManagerServiceImpl implements GroupManagerService {
+    private static Logger logger = LoggerFactory.getLogger(GroupManagerServiceImpl.class);
+    private final Channel channel;
+    private final GroupServiceFactory groupServiceFactory;
+    private ConcurrentHashMap<Integer, GroupService> groupIdToService = new ConcurrentHashMap<>();
+    private ConcurrentHashMap<String, List<String>> nodeToGroupIDList = new ConcurrentHashMap<>();
+    private final BlockNumberMessageDecoder blockNumberMessageDecoder;
+    private Client groupInfoGetter;
+    // TODO: get the fetchGroupListIntervalMs from the configuration
+    private long fetchGroupListIntervalMs = 60000;
+    private AtomicBoolean running;
+    private Thread worker;
+
+    public GroupManagerServiceImpl(Channel channel) {
+        this.channel = channel;
+        this.groupServiceFactory = new GroupServiceFactory();
+        this.groupInfoGetter = Client.build(channel);
+        fetchGroupList();
+        this.blockNumberMessageDecoder = new BlockNumberMessageDecoder(channel.getVersion());
+        this.start();
+    }
+
+    @Override
+    public Channel getChannel() {
+        return this.channel;
+    }
+
+    protected void finalize() {
+        stop();
+    }
+
+    /** Stop group list fetching thread */
+    protected void stop() {
+        if (running.get()) {
+            logger.warn("GroupManagerService has already been stopped, ");
+            return;
+        }
+        running.set(false);
+        this.worker.interrupt();
+    }
+
+    /** start the thread to obtain group list information periodically */
+    protected void start() {
+        if (this.worker != null) {
+            logger.warn("GroupManagerService has already been started");
+        }
+        this.worker =
+                new Thread(
+                        new Runnable() {
+                            @Override
+                            public void run() {
+                                logger.debug(
+                                        "start GroupManagerService: fetch and update groupInfo");
+                                while (running.get() && !Thread.currentThread().isInterrupted()) {
+                                    fetchGroupList();
+                                    try {
+                                        Thread.currentThread().sleep(fetchGroupListIntervalMs);
+                                    } catch (InterruptedException e) {
+                                        logger.warn(
+                                                "GroupManagerSerivce: capature InterruptedException: {}",
+                                                e.getMessage());
+                                        break;
+                                    }
+                                }
+                            }
+                        });
```

Comment:
Recommend use ScheduledThreadPoolExecutor to manage a schedule task.

Reply:
done.